test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -83,47 +183,27 @@ class RegAccess:
     and read their response
     """
 
-    PACKET_TYPE_REG                          = 0b00
-    PACKET_TYPE_PLAIN                        = 0b01
-    PACKET_TYPE_EVENT                        = 0b10
-    
-    PACKET_SUBTYPE_REQ_READ_REG_16           = 0b0000
-    PACKET_SUBTYPE_REQ_READ_REG_32           = 0b0001
-    PACKET_SUBTYPE_REQ_READ_REG_64           = 0b0010
-    PACKET_SUBTYPE_REQ_READ_REG_128          = 0b0011
-    PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16  = 0b1000
-    PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32  = 0b1001
-    PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64  = 0b1010
-    PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128 = 0b1011
-    PACKET_SUBTYPE_RESP_READ_REG_ERROR       = 0b1100
-    PACKET_SUBTYPE_REQ_WRITE_REG_16          = 0b0100
-    PACKET_SUBTYPE_REQ_WRITE_REG_32          = 0b0101
-    PACKET_SUBTYPE_REQ_WRITE_REG_64          = 0b0110
-    PACKET_SUBTYPE_REQ_WRITE_REG_128         = 0b0111
-    PACKET_SUBTYPE_REQ_WRITE_REG_SUCCESS     = 0b1110
-    PACKET_SUBTYPE_REQ_WRITE_REG_ERROR       = 0b1111
-
     @cocotb.coroutine
-    def send_packet(self, dut, DiPacket):
+    def send_packet(self, dut, packet):
```

Comment:
Why are the send_packet and receive_packet functions part of the RegAccess class? Is sending and receiving packets not a function of the NoC? Hence these methods belong into the NocDriver class.